### PR TITLE
[prometheus-adapter] Support environment variables (#2842)

### LINKED ISSUE #2842: [prometheus-adapter] Add option to add environment variables to the prometheus-adapter container
### Is your feature request related to a problem ?

We use a hosted prometheus in grafana cloud and want to connect to it with basic authentication.
This is possible through passing the[ --prometheus-header](https://github.com/kubernetes-sigs/prometheus-adapter/issues/439#issuecomment-896730036) flag to the adapter,  which can be achieved by setting the extraArguments variable of the values.yaml 

The problem is that there is no way to inject an environment variable to the container

This means that i cannot use gitops to deploy this chart as i need to have the password hardcoded in the values.yaml

### Describe the solution you'd like.

Have an option to inject environment variables from secrets to this container and use them as extra arguments

### Describe alternatives you've considered.

I considered to create a kubeconfig file which contains the username and password

```
apiVersion: v1
clusters:
- name: prom
  cluster:
    server: http://hosted-prometheus.com:9090
contexts:
- name: prom-context
  context:
    cluster: prom
    user: prom-user
current-context: prom-context
users:
- name: prom-user
  user:
    username: admin
    password: password123
```

then i could add it as secret and mount this to a file which can be referenced by the prometheus-auth-config flag in the adapter

Evrything would be in place for this, because i can use extraVolumes and extraVolumeMounts

I have two problems:

1. it is complex
2. Per https://github.com/kubernetes-sigs/prometheus-adapter/issues/439#issuecomment-896718145 this was not possible

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by zeritti**
Signed-off-by: zeritti <47476160+zeritti@users.noreply.github.com>

#### What this PR does / why we need it
This PR adds support for setting environment variables loaded from a secret or a config map.

#### Which issue this PR fixes

- fixes #2842

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-adapter/values.yaml [hunk 1]
```diff
@@ -151,6 +151,14 @@ tls:
   certificate: |-
     # Public key of the APIService
 
+# Set environment variables from given secrets or configmaps
+envFrom: []
```
Comment (monotek) [inline charts/prometheus-adapter/values.yaml:155]: can we just use "env" please.
it's mor flexible and can refetence configmaps and secrets too.
https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#define-container-environment-variables-using-configmap-data
https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables

File: charts/prometheus-adapter/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus-adapter
-version: 3.4.2
+version: 3.4.3
```
Comment (monotek) [inline charts/prometheus-adapter/Chart.yaml:3]: new feature should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/values.yaml (hunk 1 above)
Comment (zeritti) [reply charts/prometheus-adapter/values.yaml]: Yes, changed.

File: charts/prometheus-adapter/Chart.yaml (hunk 2 above)
Comment (zeritti) [reply charts/prometheus-adapter/Chart.yaml]: Minor number updated.

File: charts/prometheus-adapter/templates/deployment.yaml [hunk 3]
```diff
@@ -44,6 +44,10 @@ spec:
       - name: {{ .Chart.Name }}
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
+        {{- if .Values.env }}
```
Comment (monotek) [inline charts/prometheus-adapter/templates/deployment.yaml:47]: please use `{{- with .Values.env }}`

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-adapter/templates/deployment.yaml (hunk 3 above)
Comment (zeritti) [reply charts/prometheus-adapter/templates/deployment.yaml]: Changed. I preferred to stick with the conditional as it is quite common in this chart.

**Review by monotek — APPROVED** [review]

**Pull request merged**